test/thermo/PengRobinsonMFTP_Test.cpp

```diff
@@ -0,0 +1,199 @@
+    double dx = (xmax-xmin)/(numSteps-1);
+
+    for(int i=0; i < 9; ++i)
+    {
+        const double r = xmin + i*dx;
+        set_r(r);
+        test_phase->getActivityCoefficients(&activityCoeffs[0]);
+        standardConcs[0] = test_phase->standardConcentration(0);
+        standardConcs[2] = test_phase->standardConcentration(2);
+        test_phase->getActivityConcentrations(&activityConcentrations[0]);
+
+        EXPECT_NEAR(standardConcs[0] * r * activityCoeffs[0], activityConcentrations[0], 1.e-6);
+        EXPECT_NEAR(standardConcs[2] * (1-r) * activityCoeffs[2], activityConcentrations[2], 1.e-6);
+    }
+}
+
+TEST_F(PengRobinsonMFTP_Test, setTP)
+{
+    // Check to make sure that the phase diagram is accurately reproduced for a few select isobars
+
+    // All sub-cooled liquid:
+    const double p1[6] = {
+		1.666169531425807e+002,
+		1.610359415729891e+002,
+		1.561558755014150e+002,
+		1.518163520394436e+002,
+		1.479077426738997e+002,
+		1.443514857504623e+002		
+    };
+    // Phase change between temperatures 4 & 5:
+    const double p2[6] = {
+		7.085417135252475e+002,
+		6.715912546208409e+002,
+		6.252218774469104e+002,
+		5.555249258326570e+002,
+		3.218596258615833e+002,
+		2.791272952614171e+002
+    };
+    // Supercritical; no discontinuity in rho values:
+    const double p3[6] = {
+		7.695965571714201e+002,
+		7.455746209641901e+002,
+		7.196124816622006e+002,
+		6.912592525839008e+002,
+		6.598814961519255e+002,
+		6.245690038776825e+002
+    };
+
+    for(int i=0; i<6; ++i)
+    {
+        const double temp = 294 + i*2;
+        set_r(0.99);
+        test_phase->setState_TP(temp, 5542027.5);
+        EXPECT_NEAR(test_phase->density(),p1[i],1.e-8);
+
+        test_phase->setState_TP(temp, 7389370.);
+        EXPECT_NEAR(test_phase->density(),p2[i],1.e-8);
+
+        test_phase->setState_TP(temp, 9236712.5);
+        EXPECT_NEAR(test_phase->density(),p3[i],1.e-8);
+    }
+}
+
+TEST_F(PengRobinsonMFTP_Test, getPressure)
+{
+	// Check to make sure that the P-R equation is accurately reproduced for a few selected values
+
+	double a_coeff = 3.958134E+5;
+	double b_coeff = 26.6275/1000;
+	double acc_factor = 0.228;
+	double pres_theoretical, kappa, alpha, mv;
+	const double Tcrit = test_phase->critTemperature();
+
+	//Calculate kappa value
+	if (acc_factor <= 0.491) {
+		kappa = 0.37464 + 1.54226*acc_factor - 0.26992*acc_factor*acc_factor;
+	}
+	else {
+		kappa = 0.379642 + 1.487503*acc_factor - 0.164423*acc_factor*acc_factor + 0.016666*pow(acc_factor,3);
+	}
+
+	for (int i = 0; i<10; ++i)
+	{
+		const double temp = 294 + i * 2;
+		set_r(0.99);		
+		test_phase->setState_TR(temp, 1.0737);
+		mv = 1 / 1.0737 * test_phase->meanMolecularWeight();
+		//Calculate pressure using Peng-Robinson EoS
+		alpha = pow(1 + kappa*(1 - sqrt(temp / Tcrit)), 2);
+		pres_theoretical = GasConstant*temp / (mv - b_coeff) - a_coeff*alpha / (mv*mv + 2*b_coeff*mv - b_coeff*b_coeff);
+		EXPECT_NEAR(test_phase->pressure(), pres_theoretical, 10);
```

Comment:
Okay, last comment - you're requiring the model calculate a pressure within 10 Pa of the theoretical model.  Can we reduce the error tolerance, here?  Does the test fail if we go to `EXPECT_NEAR(test_phase->pressure(), pres_theoretical, 1.0);`, for instance?  If so, any idea why?

Reply:
The test fails with such a tight tolerance. The error tolerance of 1 allows only 1 Pa difference between the calculated and theoretical values.  Since the absolute pressure values are of the order 1e5 (and higher), it is reasonable to put tolerance as 10 Pa.

Reply:
We are setting CO2 mole fraction as 0.9. That is another reason why the test fails with tighter tolerances. I have modified the mole fraction to 0.999 and the tolerance to 2 Pa.